src/fdb5/database/BaseArchiveVisitor.h

```diff
@@ -32,23 +32,23 @@ class BaseArchiveVisitor : public WriteVisitor {
 
 public: // methods
 
-    BaseArchiveVisitor(Archiver &owner, const Key &field);
+    BaseArchiveVisitor(Archiver &owner, const Key& initialFieldKey);
 
 protected: // methods
 
-    virtual bool selectDatabase(const Key &key, const Key &full);
+    virtual bool selectDatabase(const Key& dbKey, const TypedKey& fullComputedKey);
 
-    virtual bool selectIndex(const Key &key, const Key &full);
+    virtual bool selectIndex(const Key& idxKey, const TypedKey& fullComputedKey);
 
-    virtual void checkMissingKeys(const Key &full);
+    virtual void checkMissingKeys(const TypedKey& fullComputedKey);
 
     virtual const Schema& databaseSchema() const;
 
     fdb5::DB* current() const;
 
 protected: // members
 
-    const Key& field_;
+    const Key& initialFieldKey_;
```

Comment:
I get scared when I see `const ref` members, on top of that `protected`
why not `const Key` and `const Key& field() {}` ?

Reply:
attention here -> https://github.com/ecmwf/fdb/actions/runs/10789281798/job/29921957489
`16 - test_fdb5_type_toKey (SEGFAULT)`

Reply:
funny enough single char fix as `const Key initialFieldKey_;` ...